### DISCUSSION TITLE
Duplex printing errors.

### DISCUSSION THREAD

**Discussion by DimulyaPlay** [sumatrapdfreader/sumatrapdf#3623] [Q&A]
Hello everyone. I am use  Sumatra for command line printing. I found that duplex parameter not working with some printers. Its not working even when default printer setting is duplex. Documents always prints in simplex mode only. I found this error on Pantum PB5100DN and Brother HL5100DN. Noticed same pattens in their model names, maybe it is problem in their OEM base drivers idk, just suggetion. Is there solution for such problems?

**Comment by DimulyaPlay** [comment]
Found that if i use -print-dialog -exit-when-done parameters, duplex works fine on that printers...  Мy regular string like '{sumatra_path} -print-to "{currentprinter}" -print-settings "bin={trayName}{duplex},fit,{copies}x" "{filepath}"' still not working with that printers

**Reply by GitHubRulesOK** [reply]
perhaps just a typo but you have no comma between `{trayName}{duplex}` and bin can be tricky to define so would put that last

-print-settings "duplex,fit,{copies}x,bin={trayName}"

**Reply by DimulyaPlay** [reply]
It works fine, I just set duplex variable from users choise like that:
    if duplexMode == 0:
        duplex = ',simplex'
    elif duplexMode == 1:
        duplex = ',duplexlong'
    elif duplexMode == 2:
        duplex = ',duplexshort'
    elif duplexMode == 3:
        duplex = ''
